Migrate existing settings to new Design

zeitschlag opened the pull request:
This is the first part for IOS-14. It migrates the existing settings over to new design. And it looks like this:

<img src="https://github.com/mastodon/mastodon-ios/assets/2580019/17565c0e-2591-4a6f-a370-3cc1540ca276" width="800">

zeitschlag committed:
Move settings-model to their appropriate location (IOS-14)

zeitschlag committed:
Remove debug-menu (IOS-14)

zeitschlag committed:
Refactor Settings-presentation and add new viewControllers (IOS-14)

zeitschlag committed:
[WIP] Add Entries for SettingsViewController (IOS-14)

zeitschlag committed:
Style settings-cell (IOS-14)

zeitschlag committed:
Turn on logout (IOS-14)

zeitschlag committed:
Fix icon-size (IOS-14)

zeitschlag committed:
Add localization to general-settings-screen (IOS-14)

zeitschlag committed:
Move coordinator-protocol into its own file (IOS-14)

zeitschlag committed:
Refactor Settings-overview (IOS-14)

zeitschlag committed:
Implement links on about-mastodon-screen (IOS-14)

zeitschlag committed:
Add localization to about-screen (IOS-14)

zeitschlag committed:
Purge cache (IOS-14)

zeitschlag committed:
Fix string (IOS-14)

zeitschlag committed:
Set title for about-screen (IOS-14)

zeitschlag committed:
Remove obsolete code

zeitschlag committed:
Fix build (IOS-14)

zeitschlag committed:
Fix build (again) (IOS-14)

zeitschlag committed:
Add mastodon-version to about-screen (IOS-14)

zeitschlag committed:
Show accountname on logout-button (IOS-14)

zeitschlag committed:
Add first draft of GeneralSettings-Screen (IOS-14)

Setting the Appearance already works, but that's it for now.

zeitschlag committed:
Don't change icon-color (IOS-14)

zeitschlag committed:
Show general-settings (IOS-14)

zeitschlag committed:
Remove duplicated strings (IOS-14)

zeitschlag committed:
Implement toggle-cell and improve selection-cell for general-settings (IOS-14)

and hook up the cell with the viewmodel.

Also: Put DiffableTableViewDataSource into its own file (and fix a typo while doing so)

zeitschlag committed:
Persist general settings (IOS-14, IOS-166)

zeitschlag committed:
Fix app.json-JSON :facepalm: (IOS-14)

zeitschlag committed:
Localize General-Setting-Screen (IOS-14)

zeitschlag committed:
Refactor Toggle-cell to be (more) reusable (IOS-14)

zeitschlag committed:
Clean up settings (a bit, IOS-14)

zeitschlag committed:
Implement basic UI for notification-settings-screen (IOS-14)

No functionality (yet), but all relevant classes and protocols are there.

zeitschlag committed:
Fix build (IOS-14)

zeitschlag committed:
Sprinkle in some localizations on Notificaiton-Screen (IOS-14)

zeitschlag committed:
Store animation-settings in UserDefaults, too (IOS-14)

Because why not?

zeitschlag committed:
Show list of Notification-policies (IOS-14)

zeitschlag committed:
Read notification-setting from CoreData (IOS-14)

aka subscription

zeitschlag committed:
Set toggle-values for notification-settings (IOS-14)

zeitschlag committed:
Set notification-types (IOS-14)

zeitschlag committed:
Toggle alerts on tap (IOS-14)

This feels kind of good :D

zeitschlag committed:
Disable notifications if they aren't enabled (IOS-14)

zeitschlag committed:
Move files

zeitschlag committed:
Add a hint in case notifications are disabled (IOS-14)

zeitschlag committed:
Sprinkle in some localization (IOS-14)

zeitschlag committed:
Disable "Support"-setting for now (IOS-14)

zeitschlag committed:
Animate toggle-switch (IOS-14)

zeitschlag committed:
Minor Cleanup (IOS-14)

zeitschlag committed:
Remove old settings

DK1DA commented on Mastodon/Scene/Settings/Notification Settings/Cells/NotificationSettingsDisabledTableViewCell.swift:
```diff
@@ -0,0 +1,67 @@
+
+    private func setupConstraints() {
+        let constraints: [NSLayoutConstraint] = [
+            appBadgeImageView.topAnchor.constraint(equalTo: contentView.topAnchor, constant: 16),
```
I feel like we should offload the `8` and `16` into some constants enum etc 😅

DK1DA replied:
But we can do this later, no need to touch all those places again right now.

DK1DA commented on Podfile:
```diff
@@ -17,8 +17,6 @@ target 'Mastodon' do
   pod 'Kanna', '~> 5.2.2'
   pod 'Sourcery', '~> 1.9'
 
-  # DEBUG
```
👍

DK1DA commented on MastodonSDK/Sources/CoreDataStack/Entity/App/Setting.swift:
```diff
@@ -14,6 +14,7 @@ public final class Setting: NSManagedObject {
     @NSManaged public var userID: String
 
 //    @NSManaged public var appearanceRaw: String
```
Shouldn't we get rid of this as well?

zeitschlag replied:
Oh, yeah, you're right. Good point! 👍

kimar reviewed (approved):
LGTM 👍  Very nice

zeitschlag committed:
Update CoreData-model to remove old settings (IOS-14)

They're stored in UserDefaults and this is kind of redundant. So: Bye!

Thanks @kimar

zeitschlag committed:
Put some constants into an extension (IOS-14)

zeitschlag committed:
Set CoreData-version

:facepalm:
